api/config.py

```diff
@@ -42,49 +52,100 @@ def get_secret(secret_name: str, is_json: bool = False) -> Optional[str]:
     try:
         get_secret_value_response = client.get_secret_value(SecretId=secret_name)
     except ClientError as e:
+        error_code = e.response["Error"]["Code"]
+        print(f"CONFIG.PY: ClientError occurred: {error_code}")
+        print(f"CONFIG.PY: Full error: {str(e)}")
         if e.response["Error"]["Code"] == "DecryptionFailureException":
             # Secrets Manager can't decrypt the protected secret text using the provided KMS key.
             # Deal with the exception here, and/or rethrow at your discretion.
+            print(
+                "CONFIG.PY: DecryptionFailureException - Secrets Manager can't decrypt the protected secret text"
+            )
             raise e
         elif e.response["Error"]["Code"] == "InternalServiceErrorException":
             # An error occurred on the server side.
             # Deal with the exception here, and/or rethrow at your discretion.
+            print(
+                "CONFIG.PY: InternalServiceErrorException - An error occurred on the server side"
+            )
             raise e
         elif e.response["Error"]["Code"] == "InvalidParameterException":
             # You provided an invalid value for a parameter.
             # Deal with the exception here, and/or rethrow at your discretion.
+            print("CONFIG.PY: InvalidParameterException - Invalid parameter value")
             raise e
         elif e.response["Error"]["Code"] == "InvalidRequestException":
             # You provided a parameter value that is not valid for the current state of the resource.
             # Deal with the exception here, and/or rethrow at your discretion.
+            print("CONFIG.PY: InvalidRequestException - Invalid request")
             raise e
         elif e.response["Error"]["Code"] == "ResourceNotFoundException":
             # We can't find the resource that you asked for.
             # Deal with the exception here, and/or rethrow at your discretion.
+            print("CONFIG.PY: ResourceNotFoundException - Can't find the secret")
             raise e
         else:
+            print(f"CONFIG.PY: Unknown ClientError: {error_code}")
             raise e
+    except Exception as e:
+        print(
+            f"CONFIG.PY: Unexpected error getting secret: {type(e).__name__}: {str(e)}"
+        )
+        raise
     else:
         # Decrypts secret using the associated KMS CMK.
         # Depending on whether the secret is a string or binary, one of these fields will be populated.
         if "SecretString" in get_secret_value_response:
-            return parse_key(
+            result = parse_key(
```

Comment:
out of curiosity, why is it better to store the result in a variable and then return it vs returning the result directly?